src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -258,13 +262,38 @@ func (s *service) Bootstrapped(ctx thrift.Context) (*rpc.NodeBootstrappedResult_
 }
 
 func (s *service) Query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryResult_, error) {
+	var (
+		ctx context.Context
+		sp  opentracing.Span
+		ok  bool
+	)
+
+	ctx = tchannelthrift.Context(tctx)
+	ctx, sp, ok = ctx.StartTraceSpan(tracepoint.QuerySpanID)
+	if ok {
+		sp.LogFields(
+			opentracinglog.String("query", req.Query.String()),
+			opentracinglog.String("namespace", req.NameSpace),
+			xopentracing.Time("start", time.Unix(0, req.RangeStart)),
+			xopentracing.Time("end", time.Unix(0, req.RangeStart)),
+		)
+	}
+
+	result, err := s.query(ctx, req)
+	if err != nil {
+		sp.LogFields(opentracinglog.Error(err))
+	}
+	sp.Finish()
```

Comment:
nit: probably easiest to follow the `defer sp.Finish()` pattern. i.e. issue the defer right after constructing the span.

Reply:
In this method since it is not in the super hot path that makes sense.  If this was a hot high frequency path I would say avoiding defer is a good idea.